source/engine/physics2/jolt/ConstraintManager.cpp

```diff
@@ -0,0 +1,219 @@
+#include "ConstraintManager.h"
+
+#include "Jolt/Jolt.h"
+#include "Jolt/Physics/PhysicsSystem.h"
+#include "Jolt/Physics/Constraints/TwoBodyConstraint.h"
+
+namespace nc::physics
+{
+auto ConstraintManager::AddConstraint(const ConstraintInfo& createInfo,
```

Comment:
I'm not a huge fan of the fact that a second body is optional, but the caller is responsible for making a dummy body to represent the "world" constraint.

How about an overload taking in a single body that sets the second body to `&JPH::Body::sFixedToWorld` under the hood?

Reply:
That seems fair, `RigidBody` already is set up that way.